packages/marketplace/src/components/ui/client-app-detail/client-app-install-confirmation.tsx

```diff
@@ -0,0 +1,167 @@
+  dispatch: Dispatch<any>,
+  setIsSuccessAlertVisible: (isVisible: boolean) => void,
+  closeInstallConfirmationModal: () => void,
+) => {
+  return () => {
+    dispatch(
+      clientFetchAppDetail({
+        id: appId,
+        clientId,
+      }),
+    )
+    closeInstallConfirmationModal()
+    setIsSuccessAlertVisible(true)
+  }
+}
+
+export const handleInstallButtonClick = (
+  appId: string,
+  clientId: string,
+  dispatch: Dispatch<any>,
+  setIsSuccessAlertVisible: (isVisible: boolean) => void,
+  closeInstallConfirmationModal: () => void,
+) => {
+  return () => {
+    dispatch(
+      appInstallationsRequestInstall({
+        appId,
+        callback: handleInstallAppSuccessCallback(
+          appId,
+          clientId,
+          dispatch,
+          setIsSuccessAlertVisible,
+          closeInstallConfirmationModal,
+        ),
+      }),
+    )
+  }
+}
+
+export const handleSuccessAlertButtonClick = history => {
+  return () => {
+    history.replace(routes.CLIENT)
+  }
+}
+
+export const handleSuccessAlertMessageAfterClose = (setIsSuccessAlertVisible: (isVisible: boolean) => void) => {
+  return () => {
+    setIsSuccessAlertVisible(false)
+  }
+}
+
+const ClientAppInstallConfirmation: React.FC<ClientAppInstallConfirmationProps> = ({
+  appDetailData,
+  visible,
+  closeInstallConfirmationModal,
+}) => {
+  const history = useHistory()
+  const [isSuccessAlertVisible, setIsSuccessAlertVisible] = React.useState(false)
+  const clientId = useSelector(selectClientId)
+  const installationFormState = useSelector(selectInstallationFormState)
+  const isSubmitting = installationFormState === 'SUBMITTING'
+
+  const { name, id = '', scopes = [] } = appDetailData || {}
+
+  const dispatch = useDispatch()
+  const onSuccessAlertButtonClick = React.useCallback(handleSuccessAlertButtonClick(history), [history])
+
+  return (
+    <>
+      <Modal
+        visible={visible}
+        title={`Confirm ${name} installation`}
+        afterClose={closeInstallConfirmationModal}
+        footerItems={
+          <>
+            <Button
+              dataTest="agree-btn"
+              loading={isSubmitting}
+              className={appPermissionContentStyles.installButton}
+              type="button"
+              variant="primary"
+              onClick={handleInstallButtonClick(
+                id,
+                clientId,
+                dispatch,
+                setIsSuccessAlertVisible,
+                closeInstallConfirmationModal,
+              )}
+            >
+              Confirm
+            </Button>
+            <Button
+              dataTest="disagree-btn"
+              disabled={false}
+              className={appPermissionContentStyles.installButton}
+              type="button"
+              variant="danger"
+              onClick={closeInstallConfirmationModal}
+            >
+              Cancel
+            </Button>
+          </>
+        }
+      >
+        <>
+          {scopes.length ? (
```

Comment:
I think should move this out because function render too long.

Reply:
Hi @tanphamhaiduong,
Because this view is going to be changed when building the new UI for standalone app page, refactoring these codes here will means nothing, we will still have to rework anyway